### ISSUE TITLE
identity loss for cyclegan

### ISSUE THREAD

**Issue by cdlyr** [junyanz/pytorch-CycleGAN-and-pix2pix#1159]
https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/c94161791aaf050dfffb4f2b2fac815997b37aa6/models/cycle_gan_model.py#L36

Hi,

I purposely increased the Identity loss for my own project and the results got worse after the change. I took a look at the code carefully and then realized the Identity loss may be not correct. Minimizing ||G_A(B) - B|| and ||G_B(A) - A|| will just make the two generators have no effect to the input as much as possible. It's supposed to be
 `Identity loss (optional): lambda_identity * (||G_A(A) - B|| * lambda_B + ||G_B(B) - A|| * lambda_A) ` 
instead if I understand properly. Please let me know if it is an issue. 
Thank you.

**Comment by JunlinHan** [comment]
G_A is G(A->B). If input is already B, B->B should keep B unchanged. That's the purpose of identity loss.
For your understanding, it seems make sense. But remember we are using unpaired data, A->G_A(A) has no corresponding B.
Hope my reply helps.

**Comment by cdlyr** [comment]
Thanks for the reply! I had this issue because I switched the training set into aligned data and dramatically increase the weight for identity loss. Generally for unpaired data I think the original loss can help the model to learn sort of structure similarity even the mapping is not exactly what we want, which makes sense to me now.
